packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -786,6 +804,11 @@ contract ControllerHelper is UniswapControllerHelper, AaveControllerHelper, IERC
                 ControllerHelperDiamondStorage.getAddressAtSlot(3),
                 _amountToPay
             );
+        } else if (
+            ControllerHelperDataType.CALLBACK_SOURCE(_callSource) ==
+            ControllerHelperDataType.CALLBACK_SOURCE.GENERAL_SWAP
+        ) {
+            IERC20(_tokenIn).transfer(ControllerHelperDiamondStorage.getAddressAtSlot(3), _amountToPay);
```

Comment:
this only works for WETH/ETH to oSQTH swaps when we have a WETH balance in the contract

any payable ETH can't be used for this as it isn't ever wrapped and any withdrawn ETH from a vault can't be used for it as it is withdrawn as ETH.

only WETH from an existing uniswap position could be used for this.  One potential solution is wrapping ETH balance before the callback and unwrapping weth after transfering the amount owed from the flashswap.

Reply:
Another potential solution is to send some ETH to uniswap as part of the call or repay with ETH, this possible?

Reply:
I will just wrap the contract balance again, also to be consistent by repaying WETH to pool (not sending ETH)